tests/test_frontdoor.py

```diff
@@ -1473,3 +1482,57 @@ def test_on_check_answers_page_changing_to_social_housing_hides_park_home_questi
 
     _assert_change_button_is_hidden(page, "park-home")
     _assert_change_button_is_hidden(page, "park-home-main-residence")
+
+
+@unittest.mock.patch("help_to_heat.frontdoor.interface.EPCApi", MockEPCApi)
+@pytest.mark.parametrize("park_home", [True, False])
+def test_on_check_answers_page_if_park_home_main_residence_is_hidden_depending_on_park_home_answer(park_home):
+    _check_page, page, session_id = setup_client_and_page()
+
+    # Answer main flow
+    page = _answer_house_questions(page, session_id, benefits_answer="Yes", park_home=park_home)
+
+    # press back to get to summary page
+    page = page.click(contains="Back")
+    assert page.has_one("h1:contains('Check your answers')")
+
+    if park_home:
+        _assert_change_button_is_not_hidden(page, "park-home-main-residence")
+    else:
+        _assert_change_button_is_hidden(page, "park-home-main-residence")
+
+
+@unittest.mock.patch("help_to_heat.frontdoor.interface.EPCApi", MockEPCApi)
+def test_on_check_answers_page_changing_to_no_loft_hides_loft_follow_up_questions():
+    _check_page, page, session_id = setup_client_and_page()
+
+    # Answer main flow
+    page = _answer_house_questions(page, session_id, benefits_answer="Yes", has_loft=True)
+
+    # press back to get to summary page
+    page = page.click(contains="Back")
+    assert page.has_one("h1:contains('Check your answers')")
+
+    _assert_change_button_is_not_hidden(page, "loft-access")
+    _assert_change_button_is_not_hidden(page, "loft-insulation")
+
+    page = _click_change_button(page, "loft")
+    assert page.has_text("Do you have a loft that has not been converted into a room?")
+
+    page = _check_page(page, "loft", "loft", "No, I do not have a loft or my loft has been converted into a room")
+    assert page.has_one("h1:contains('Check your answers')")
+
+    _assert_change_button_is_hidden(page, "loft-access")
+    _assert_change_button_is_hidden(page, "loft-insulation")
+
+
+def setup_client_and_page():
```

Comment:
```suggestion
def _setup_client_and_page():
```
By convention since this is a 'private' method.